fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvPartitionCompensator.java

```diff
@@ -524,9 +524,9 @@ public static ScalarOperator compensateQueryPartitionPredicate(MaterializationCo
             if (partitionPredicate == null) {
                 logMVRewrite(mvContext.getMv().getName(), "Compensate partition failed for scan {}",
                         scanOperator.getTable().getName());
-                return null;
```

Comment:
1. We may check why `partitionPredicate` returns null which should be the root cause.
2. For each scan operator, the partition predicate should be compensated no matter `isCompensatePartition` is true or false.

Reply:
You can refer `MvRewriteJoinTest` add a unit test?

Reply:
return null happend when a sql like `SELECT ... where id in (select id from test_alias)`, test_alias has no partition predicate condition